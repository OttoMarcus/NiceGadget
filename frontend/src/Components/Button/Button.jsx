import React from "react";
import PropTypes from "prop-types";
import styles from "./Button.module.scss";

const Button = ({ onClick, children, backgroundColor }) => {
  const buttonStyle = {
    backgroundColor,
  };

  return (
    <button style={buttonStyle} className={styles.button} onClick={onClick}>
      {children}
    </button>
  );
};

Button.propTypes = {
  onClick: PropTypes.func,
  children: PropTypes.node.isRequired,
  backgroundColor: PropTypes.string,
};

Button.defaultProps = {
  onClick: () => {},
  backgroundColor: "#905BFF",
};

export default Button;
