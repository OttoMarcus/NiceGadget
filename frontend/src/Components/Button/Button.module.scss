@import "../../styles/abstracts/variables";
@import "../../styles/abstracts/mixins";

.button {
  @include flex-center;
  @include button;
  height: 40px;
  width: 100%;
  background-color: $ui-color-primary-accent;
  color: $ui-color-gray-white;
  border: none;
  margin-right: 8px;

  &:disabled {
    background-color: $ui-color-gray-surface-1;
  }

  &:hover {
    background-color: $ui-color-primary-accent-hover;
  }

  &:active {
    background-color: $ui-color-gray-selected-surface-2;
  }
}
