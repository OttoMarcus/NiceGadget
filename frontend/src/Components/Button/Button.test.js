import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";

import Button from "./Button";

describe("Button tests: ", () => {
  test("should be clickable and have text", () => {
    const btn = render(<Button btnName={"Click me"} />);

    expect(btn).toMatchSnapshot();
  });
});
