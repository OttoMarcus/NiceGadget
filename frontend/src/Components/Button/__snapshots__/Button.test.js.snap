// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="button"
    style="background-color: rgb(144, 91, 255);"
  >
    Click me
  </button>
</DocumentFragment>
`;
